### ISSUE TITLE
TypeError: Module '"node-appwrite"' has no exported member 'InputFile'

### ISSUE THREAD

**Issue by Vicidevs** [adrianhajdin/healthcare#16]
Cannot find the InputFile, any alternative?

**Comment by bharatpaliwal-169** [comment]
> use import { InputFile } from 'node-appwrite/file';

and use InputFile.fromBuffer().

**Comment by biyiemmy** [comment]
> use import { InputFile } from 'node-appwrite/file';

this seems not to be working from my end

**Comment by KelberStuchi** [comment]
import { InputFile } from 'node-appwrite/file';

let file;
    if (identificationDocument) {
      const inputFile =
        identificationDocument &&
        InputFile.fromBuffer(
          identificationDocument?.get("blobFile") as Blob,
          identificationDocument?.get("fileName") as string
        );

      file = await storage.createFile(BUCKET_ID!, ID.unique(), inputFile);
    }

**Comment by biyiemmy** [comment]
> import { InputFile } from 'node-appwrite/file';
> 
> let file; if (identificationDocument) { const inputFile = identificationDocument && InputFile.fromBuffer( identificationDocument?.get("blobFile") as Blob, identificationDocument?.get("fileName") as string ); file = await storage.createFile(BUCKET_ID!, ID.unique(), inputFile); }

Hi bro
can you share the full code, it seems Im not getting you

**Comment by TkRj** [comment]
I encountered an import error for 'fs' module for the InputFile module. This is due to fs not being available in the browser as it only exists in nodejs environment. If you face the same issue,  go to **next.config.mjs** file and update with the following: 
<details>
<summary>Code</summary>
const nextConfig = {
  future: {
    webpack5: true,
  },
  webpack(config) {
    config.resolve.fallback = {
      ...config.resolve.fallback, 
      fs: false,
    };

    return config;
  },
};`
</details>

[source](https://stackoverflow.com/questions/67478532/module-not-found-cant-resolve-fs-nextjs/67478653#67478653)

**Comment by alfattal7** [comment]
use this const sdk = require('node-appwrite');
and befor any function of "node-appwrite" just right sdk => sdk.Query.equal("email", [user.email]) etc.

**Comment by FinzyPHINZY** [comment]
> I encountered an import error for 'fs' module for the InputFile module. This is due to fs not being available in the browser as it only exists in nodejs environment. If you face the same issue, go to **next.config.mjs** file and update with the following:
> 
> Code
> [source](https://stackoverflow.com/questions/67478532/module-not-found-cant-resolve-fs-nextjs/67478653#67478653)

This works!

**Comment by KevinFalah** [comment]
> use import { InputFile } from 'node-appwrite/file';

its worked for me, thanks

**Comment by SanadKhan** [comment]
@biyiemmy @Vicidevs 

node-appwrite13.0.0

in your utils file add
```
export const getBufferImage = async(fileInfo: File) => {
  try {
    const bytes = await fileInfo.arrayBuffer();
    const bufferImage = Buffer.from(bytes);
    return bufferImage;
  } catch (error) {
    throw error
  }
}
```
inside patient.action.ts
```
export const registerPatient = async({ identificationDocument, ...patient}: RegisterUserParams) => {
    try {
        let file        
        if(identificationDocument) {
            const docFile =  identificationDocument?.get('blobFile') as File;
            const fileName = identificationDocument?.get('fileName') as string
            const bufferImg: Buffer = await getBufferImage(docFile);
            const inputFile = InputFile.fromBuffer(bufferImg,fileName)            
            file = await storage.createFile(BUCKET_ID!, ID.unique(), inputFile)
        }
        
        const newPatient = await database.createDocument(
            DATABASE_ID!,
            PATIENT_COLLECTION_ID!,
            ID.unique(),
            { 
                identificationDocumentId: file?.$id ? file.$id : null,
                identificationDocumentUrl: file?.$id ? `${ENDPOINT}/storage/buckets/${BUCKET_ID}/files/${file?.$id}/view?project=${PROJECT_ID}` : null,
                ...patient
            } 
        )
        return parseStringify(newPatient)
        // return 
    } catch (error: any) {
        console.log("Erorr", error);
        
    }
}
```
since the `InputFile.fromBuffer()` and `InputFile.fromBlob()` were having too many issues to handle on version compatibility. Converting the Blob to Buffer and passing it to storage does the job.
If anyone has a better solution let me know :+1:

**Comment by BerlinWebDev** [comment]
Hey Guys! You just have to the file "use server" at the top of the file do get it work..

**Comment by MuhamedGamall** [comment]
change your "node-appwrite" version and add this version
"node-appwrite": "^12.0.1",
Then add this `InputFile.fromBlob()` instead of `InputFile.fromBuffer()`

**Comment by tewodros-m** [comment]
instead of InputFile.fromBuffer() use File() constructor as follows it workes for me.

// REGISTER PATIENT
export const registerPatient = async ({
  identificationDocument,
  ...patient
}: RegisterUserParams) => {
  try {
    // Upload file
    let file;
    if (identificationDocument) {
      const inputFile = new File(
        [identificationDocument?.get("blobFile") as Blob],
        identificationDocument?.get("fileName") as string
      );

      file = await storage.createFile(BUCKET_ID!, ID.unique(), inputFile);
    }

    // Create new patient document
    const newPatient = await databases.createDocument(
      DATABASE_ID!,
      PATIENT_COLLECTION_ID!,
      ID.unique(),
      {
        identificationDocumentId: file?.$id ? file.$id : null,
        identificationDocumentUrl: file?.$id
          ? `${ENDPOINT}/storage/buckets/${BUCKET_ID}/files/${file.$id}/view??project=${PROJECT_ID}`
          : null,
        ...patient,
      }
    );

    return parseStringify(newPatient);
  } catch (error) {
    console.error("An error occurred while creating a new patient:", error);
  }
};

**Comment by Rufaida-Shaik** [comment]
> change your "node-appwrite" version and add this version "node-appwrite": "^12.0.1", Then add this `InputFile.fromBlob()` instead of `InputFile.fromBuffer()`

THIS WORKED!!!!!!!!!!!!!!! i have been banging my head since two days. re wrote the code fpr multiple times. this works T T. Thanks a ton!

**Comment by theTechbug** [comment]
you need to doTypeError: Module 'node-appwrite' has no exported member 'InputFile'. This is due to the newer version of 'node-appwrite' no longer including 'InputFile'. Downgrading the 'node-appwrite' package to version 12.0.1 should resolve this issue.

**Comment by InukaAasitha** [comment]
above mentioned nothing works. somebody please help me. im doing this for my university project, time is out pleaseee.

**Comment by SanadKhan** [comment]
> you need to doTypeError: Module 'node-appwrite' has no exported member 'InputFile'. This is due to the newer version of 'node-appwrite' no longer including 'InputFile'. Downgrading the 'node-appwrite' package to version 12.0.1 should resolve this issue.

@InukaAasitha  have you tried this?

**Comment by InukaAasitha** [comment]
Yes, no luck yet. I will try copy and past whole file and try again and let you know where it works or not. Appreciate your concern! thank you so much.. @SanadKhan
